### ISSUE TITLE
[Pager] pagerState.animateScrollToPage() will tigger unnecessary recompose

### ISSUE THREAD

**Issue by GavinNovate** [google/accompanist#1403]
**Description**
```kotlin
@Composable
fun Pager() {


    Column {
        val pagerState = rememberPagerState()

        Row(
            modifier = Modifier
                .fillMaxWidth()
                .wrapContentHeight()
        ) {
            val scope = rememberCoroutineScope()
            Button(
                onClick = {
                    scope.launch {
                        pagerState.animateScrollToPage(0)
                    }
                }
            ) {
                Text(text = "Pager1")
            }
            Button(
                onClick = {
                    scope.launch {
                        pagerState.animateScrollToPage(1)
                    }
                }
            ) {
                Text(text = "Pager2")
            }
        }

        HorizontalPager(
            count = 2,
            state = pagerState,
            modifier = Modifier
                .fillMaxWidth()
                .weight(1f)
        ) { index ->
            when (index) {
                0 -> Pager1()
                1 -> Pager2()
            }
        }
    }
}


var pager1createTimes = 0
var pager1composeTimes = 0

@Composable
fun Pager1() {
    LaunchedEffect(Unit) {
        Log.d(TAG, "Pager1: create times：${++pager1createTimes}")
    }
    Log.d(TAG, "Pager1: compose times：${++pager1composeTimes}")
    Text(text = "Pager1")
}

var pager2createTimes = 0
var pager2composeTimes = 0

@Composable
fun Pager2() {
    LaunchedEffect(Unit) {
        Log.d(TAG, "Pager2: create times：${++pager2createTimes}")
    }
    Log.d(TAG, "Pager2: compose times：${++pager2composeTimes}")
    Text(text = "Pager2")
}
```

**Steps to reproduce**
1. click Pager2 Button
2. click Pager1 Button (Pager1 will recreate and recompose)
3. click Pager2 Button
4. click Pager1 Button(Pager1 will recreate and recompose)

Logs:
```
2022-11-19 14:30:50.338 22937-22937 Pager                   com.example.composedemo              D  Pager1: compose times：1
2022-11-19 14:30:50.344 22937-22937 Pager                   com.example.composedemo              D  Pager2: compose times：1
2022-11-19 14:30:50.414 22937-22937 Pager                   com.example.composedemo              D  Pager1: create times：1
2022-11-19 14:30:50.414 22937-22937 Pager                   com.example.composedemo              D  Pager2: create times：1
2022-11-19 14:31:02.023 22937-22937 Pager                   com.example.composedemo              D  Pager1: compose times：2
2022-11-19 14:31:02.036 22937-22937 Pager                   com.example.composedemo              D  Pager1: create times：2
2022-11-19 14:34:18.620 22937-22937 Pager                   com.example.composedemo              D  Pager1: compose times：3
2022-11-19 14:34:18.625 22937-22937 Pager                   com.example.composedemo              D  Pager1: create times：3
```

**Expected behavior** 
1. click Pager Button
2. click Pager1 Button should not recreate and recompose

**Additional context**
1. Swipe Pager to Pager2 then Swipe Pager to Pager1 will not recreate and recompose
2. pagerState.animateScrollToPage() & pagerState.scrollToPage() will recreate and recompose

**Comment by andkulikov** [comment]
We only compose the visible page. Which means that when you move to page 1 the previous page will not be composed anymore. Then you go back to page and we compose it again.

**Comment by GavinNovate** [comment]
When I enter the page, I am on the page 1 at this time, and the page 2 already composed, so I thought that ViewPager would cache the pages。

According to what you mean, should the second page not be composed when entering for the first time?

**Comment by andkulikov** [comment]
> When I enter the page, I am on the page 1 at this time, and the page 2 already composed, so I thought that ViewPager would cache the pages

there is no real caching. we only compose items which a visible

> According to what you mean, should the second page not be composed when entering for the first time?
Yes, it is a bug which we will fix in Compose 1.4

**Comment by andkulikov** [comment]
As I mentioned in the previous comment it is a bug fixed in Compose 1.4
